Refactor: Rename movies package to movie

skaik-mo opened the pull request:
Updated all relevant import statements and file paths across the codebase to reflect the new package name.

Added `userRating` field to `Movie` entity and updated dummy data in tests. Removed unused test cases in `GetRatedSeriesUseCaseTest`.

skaik-mo committed:
Refactor: Rename movies package to movie

Updated all relevant import statements and file paths across the codebase to reflect the new package name.

Added `userRating` field to `Movie` entity and updated dummy data in tests.
Removed unused test cases in `GetRatedSeriesUseCaseTest`.

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop:
  update imports
  rename home api module
  rename home module
  change packages for the details presentation and api
  change name space for the details presentation and api

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/presentation/details/model/MovieUi.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/castCredit/CastCreditViewModel.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/recommended/movie/RecommendedMoviesViewModel.kt

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop:
  Refactor: Update resource imports in presentation/explore
  Refactor: Update explore module namespaces

AhmedMohamed056 commented on domain/media/src/test/kotlin/com/giraffe/media/series/usecase/GetRatedSeriesUseCaseTest.kt:
```diff
@@ -25,36 +18,36 @@ class GetRatedSeriesUseCaseTest {
+//        coVerify { repository.getRatedSeries(userId) }
+//        assertEquals(expectedResult, result)
+//    }
 }
```
What's this skaik?

AhmedMohamed056 commented on domain/media/src/test/kotlin/com/giraffe/media/movie/usecase/GetRatedMoviesUseCaseTest.kt:
```diff
@@ -34,20 +34,16 @@ class GetRatedMoviesUseCaseTest {
 
         // Then
         coVerify { getUserUseCase() }
         coVerify { repository.getRatedMovies(accountId) }
```
make this test clear you test getRatedMovies or getUserUseCase

skaik-mo replied:
This PR changes the module name from `movies` to `movie`. I have another PR that will cover checking all the tests.

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop:
  resolve conflicts
  resolved conflicts.
  Refactor: Remove unused imports in HomeNavGraph.kt
  Refactor: Migrate profile module to presentation layer
  remove load movies by genres.
  remove else in show more factory.
  add  get section title to show more section type and change name to show more.
  change name to show more.
  change name to show more state.
  remove section title.
  replace occurrences of navigate section with navigate to show more.
  add show more section type to navigate to show more screen.
  remove section title.
  replaces all usages of the literal phrase navigate section with navigate to show more.
  rename movies list to show more.
  add presentation module in di for show more strategy.
  use show more section type enum instead of string.
  implement show more strategy in show more view model
  add show more strategy.
  change movie list to show more.

# Conflicts:
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/movies_list/MoviesListViewModel.kt
#	presentation/profile/src/main/java/com/giraffe/presentation/profile/screens/collections/collection/CollectionViewModel.kt

skaik-mo committed:
Refactor: Update movie use case import paths

The import paths for movie-related use cases in `ShowMoreStrategy.kt` and `presentationModule.kt` have been updated from `com.giraffe.media.movies.usecase` to `com.giraffe.media.movie.usecase`.

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop:
  fix error in seasons screen
  fix error in series details screen and seasons screen
  add nullable in release year in season ui
  resolve series details error
  refactor releaseYear in series ui model
  add nullable to releaseYear in series ui model
  make series use case more readable
  remove default value in entity
  rename getByGenre function
  fix presentation module
  refactor clean series cache use case
  remove returns not import
  merge develop
  merge develop
  refactor mapper to series entity
  - refactor series repo functions - refactor series use case - refactor series use case test
  - refactor series entity - refactor mapper for series - remove season table from room database

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/presentationModule.kt
#	domain/media/src/test/kotlin/com/giraffe/media/movie/usecase/GetRatedMoviesUseCaseTest.kt
#	domain/media/src/test/kotlin/com/giraffe/media/series/usecase/GetRatedSeriesUseCaseTest.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/show_more/ShowMoreStrategy.kt
